### PR TITLE
IDE-2487 add preferences for blade repo url and valid time

### PR DESCRIPTION
**Pull request by AndyWu2015**

### CODE REVIEW

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/BladeCLI.java [hunk 1]
```diff
@@ -270,6 +302,28 @@ private static String getLatestRemoteBladeCLIJar() throws BladeCLIException
...
+        String retVal = URLEncoder.encode( repoURL, "UTF-8" );
+
+        return new File( repoCache, retVal + "plugins" );
+    }
+
+    public static String getRepoURL()
+    {
+        IEclipsePreferences prefs = InstanceScope.INSTANCE.getNode( ProjectCore.PLUGIN_ID );
+
+        return prefs.get( "repoUrl", repoUrl );
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/BladeCLI.java]: Make these strings constants that say something like "BLADE_CLI_REPO_URL"

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/BladeCLI.java [hunk 2]
```diff
@@ -270,6 +302,28 @@ private static String getLatestRemoteBladeCLIJar() throws BladeCLIException
...
+    }
+
+    public static String getValidTime()
+    {
+        IEclipsePreferences prefs = InstanceScope.INSTANCE.getNode( ProjectCore.PLUGIN_ID );
+
+        return prefs.get( "validTime", "24h" );
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/BladeCLI.java]: Call this one "BLADE_CLI_REPO_UP2DATE_CHECK"

**Comment by gamerson** [comment]
Closing see comments, thanks!

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/BladeCLI.java (hunk 2 above)
Comment (AndyWu2015) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/BladeCLI.java]: got it

**Pull request closed without merging**